### PR TITLE
Change busy database print to only use one line

### PR DESCRIPTION
**Pull request by jackkoenig**
Instead of printing the same message every second, it now prints the
message once and just adds an additional period per second.

### CODE REVIEW

File: src/database.cpp [hunk 1]
```diff
@@ -211,10 +215,18 @@ std::string Database::open(bool wait, bool memory) {
     sqlite3_free(fail);
 
     if (!wait || ret != SQLITE_BUSY) {
+      if (waiting) {
+        std::cerr << std::endl;
+      }
       close();
       return out;
```
Comment (jackkoenig) [inline src/database.cpp:222]: I'm not sure when this branch is executed but I added it anyway 🤷

File: src/database.cpp [hunk 2]
```diff
@@ -211,10 +215,18 @@ std::string Database::open(bool wait, bool memory) {
...
     } else {
-      std::cerr << "Database wake.db is busy; waiting 1 second ..." << std::endl;
+      if (waiting) {
+        std::cerr << ".";
+      } else {
+        waiting = true;
+        std::cerr << "Database wake.db is busy; waiting 1 second ...";
```
Comment (mmjconolly) [inline src/database.cpp:228]: Maybe something like `s/waiting 1 second .../waiting ./` as we might wait longer than 1 second

File: src/database.cpp [hunk 3]
```diff
@@ -200,6 +201,9 @@ std::string Database::open(bool wait, bool memory) {
       const char *set_version = "insert or ignore into schema(version) values(" SCHEMA_VERSION ");";
       ret = sqlite3_exec(imp->db, get_version, &schema_cb, 0, 0);
       if (ret == SQLITE_OK) {
+        if (waiting) {
+          std::cerr << std::endl;
+        }
```
Comment (mmjconolly) [inline src/database.cpp:206]: What motivates this print statement?

File: src/database.cpp (hunk 2 above)
Comment (jackkoenig) [reply src/database.cpp:228]: Sounds good to me, mind phrasing it as a `suggestion`?

File: src/database.cpp (hunk 3 above)
Comment (terpstra) [reply src/database.cpp:206]: I think this needs to be move to the top of the enclosing 'of'; ie: right before the "use an empty entropy table" comment.

File: src/database.cpp (hunk 2 above)
Comment (mmjconolly) [inline src/database.cpp:228]:
```suggestion
        std::cerr << "Database wake.db is busy; waiting.";
```

Comment (jackkoenig) [reply src/database.cpp]: Done.

File: src/database.cpp (hunk 3 above)
Comment (jackkoenig) [reply src/database.cpp]: Done.

File: src/database.cpp (hunk 2 above)
Comment (jackkoenig) [reply src/database.cpp]: Oh sorry, I missed that you had done the suggestion, I made the change though.

File: src/database.cpp (hunk 1 above)
Comment (jackkoenig) [reply src/database.cpp:222]: @terpstra does this one need to be here? I'm not sure when we're returning vs. breaking from the loop.

**Comment by mmjconolly** [comment]
Just tested it, seems to work as expected
```
$ ../bin/wake --no-tty -x '"sleep 5m" | makePlan "" Nil | runJob' &
$ ../bin/wake -x '"sleep 5m" | makePlan "" Nil | runJob' 
Database wake.db is busy; waiting .......^C
```

**Review by mmjconolly — APPROVED** [review]

**Comment by terpstra** [comment]
Didn't I already review this? I don't see my comments anywhere :-/

**Review by terpstra — APPROVED** [review]
LGTM. I could swear I asked for a change to be made, but when I look over the PR now, everything seems to be correct.

**Pull request merged**